knora/dsplib/utils/xml_upload.py

```diff
@@ -910,94 +918,340 @@ def xml_upload(input_file: str, server: str, user: str, password: str, imgdir: s
         if resource.ark:
             resource_iri = convert_ark_v0_to_resource_iri(resource.ark)
 
+        # in case of a multimedia resource: upload the multimedia file
         resource_bitstream = None
         if resource.bitstream:
-            img = sipi_server.upload_bitstream(os.path.join(imgdir, resource.bitstream.value))
-            internal_file_name_bitstream = img['uploadedFiles'][0]['internalFilename']
-            resource_bitstream = resource.get_bitstream(internal_file_name_bitstream, permissions_lookup)
+            try:
+                img = try_sipi_upload(sipi_server=sipi_server, filepath=os.path.join(imgdir, resource.bitstream.value))
+                internal_file_name_bitstream = img['uploadedFiles'][0]['internalFilename']
+                resource_bitstream = resource.get_bitstream(internal_file_name_bitstream, permissions_lookup)
+            except BaseError:
+                print(f"ERROR while trying to create resource '{resource.label}' ({resource.id}). ")
+                failed_uploads.append(resource.id)
+                continue
```

Comment:
here again, all exceptions are caught, but not propagated/handled, which is bad practice.

Reply:
Ah... I'm sorry! I misread this as `BaseException`. Never mind my comment in this case. (But those two are named too similar to not mix them up...)